### PR TITLE
feat: Load mandatory data during the splash screen

### PR DESCRIPTION
**Pull request by Adaendra**
## 📖 Description
Setup mandatory data loading during the splash screen.

### ⁉️ Related Issues
#52

### 🧪 How to test the change?
Start the application, the splash screen should take a bit more of time due to the data load, but then all groups and kana are loaded and no more api call are done.

### ☑️ Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] I have added/updated tests.
- [x] Make sure to add either one of the following labels: `version: Major`,`version: Minor` or `version: Patch`.

### CODE REVIEW

File: lib/src/core/dataloaders/group_dataloader.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import 'package:kana_to_kanji/src/core/models/group.dart';
+import 'package:kana_to_kanji/src/core/services/api_service.dart';
+import 'package:kana_to_kanji/src/locator.dart';
+
+class GroupDataLoader {
+  final ApiService _apiService = locator<ApiService>();
+  final Isar _isar = locator<Isar>();
+
+  /// Load all the groups from the API.
+  Future loadCollection() async {
```
Comment (apomalyn) [inline lib/src/core/dataloaders/group_dataloader.dart:14]: Aussi est-ce que faudrait pas mettre un safeguard pour pas que le call se fasse a chaque démarrage? 🤔 
Est-ce que justement de faire ça ne overload pas la DB avec des duplicats?

Est-ce que tu veux bien m'ajouter un test pour ça, please? 😄

File: lib/src/locator.dart [hunk 2]
```diff
@@ -44,4 +46,12 @@ void setupLocator() {
       () => KanaRepository(),
       dependsOn: [Isar]);
   locator.registerSingleton<SettingsRepository>(SettingsRepository());
+
+  // Data Loaders
+  locator.registerSingletonWithDependencies<GroupDataLoader>(
+      () => GroupDataLoader(),
+      dependsOn: [Isar]);
```
Comment (apomalyn) [inline lib/src/locator.dart:53]: Simple question, est-ce que cela aurait pu etre ca a la place? 🤔 

```suggestion
  locator.registerSingletonAsync<GroupDataLoader>(
      () async {
          final instance = GroupDataLoader();
          await instance.loadCollection();
          return instance;
      },
      dependsOn: [Isar, ApiService]);
```

En soit, ce que tu as mis fonctionne très bien et en fonction de comment on va implementer la sync de données ça bougerait de là

File: lib/src/core/dataloaders/group_dataloader.dart (hunk 1 above)
Comment (Adaendra) [reply lib/src/core/dataloaders/group_dataloader.dart:14]: Actuellement il n'y a pas de base de donnée embarquée :p 

Cette vérification sera faite quand Isar 4.0 va permettre d'avoir du stockage entre deux utilisations ;)

File: lib/src/locator.dart (hunk 2 above)
Comment (Adaendra) [reply lib/src/locator.dart:53]: Si on fait ça là c'est pendant le splash screen d'android et pas le splash screen animé. (d'après ce dont on avait discuté ce weekend) Du coup c'est pour ça que je n'ai pas mis ça ici

**Review by apomalyn — APPROVED** [review]
